compiler/rustc_hir_analysis/src/collect.rs

```diff
@@ -55,7 +55,18 @@ mod type_of;
 // Main entry point
 
 fn collect_mod_item_types(tcx: TyCtxt<'_>, module_def_id: LocalModDefId) {
-    tcx.hir().visit_item_likes_in_module(module_def_id, &mut CollectItemTypesVisitor { tcx });
+    let items = tcx.hir_module_items(module_def_id);
+    let hir = tcx.hir();
+    let _ = items.par_items(|item| Ok(CollectItemTypesVisitor { tcx }.visit_item(hir.item(item))));
+    let _ = items.par_trait_items(|item| {
+        Ok(CollectItemTypesVisitor { tcx }.visit_trait_item(hir.trait_item(item)))
+    });
+    let _ = items.par_impl_items(|item| {
+        Ok(CollectItemTypesVisitor { tcx }.visit_impl_item(hir.impl_item(item)))
+    });
+    let _ = items.par_foreign_items(|item| {
+        Ok(CollectItemTypesVisitor { tcx }.visit_foreign_item(hir.foreign_item(item)))
+    });
```

Comment:
In general, I'd expect having something like `par_visit_item_likes_in_module` instead of doing things like this here or in wf checking.

Reply:
That can't be done nicely, because `visit_item_likes_in_module` uses a visitor. I guess we could add something that creates the visitor 4 times just like I did here, but considering after this PR we again have only a single use site of such a `par_visit_item_likes_in_module` method, keeping it inline at the one use site seems best to me?

Reply:
>considering after this PR we again have only a single use site of such a `par_visit_item_likes_in_module` method

If overhead of parallel queries was low enough, then I'd expect pretty much every `visit_item_likes_in_module`s and `visit_all_item_likes_in_crate`s to benefit from being turned into its `par_` version.

Reply:
I'll add it to the list of things I am investigating around the parallel compiler